### PR TITLE
remove std::move

### PR DESCRIPTION
**Pull request by StrikerRUS**
Fix warnings from Clang compilation: https://github.com/microsoft/LightGBM/pull/3443#discussion_r507728708.

### CODE REVIEW

File: src/boosting/gbdt_model_text.cpp [hunk 1]
```diff
@@ -393,7 +393,7 @@ std::string GBDT::SaveModelToString(int start_iteration, int num_iteration, int
     ss << loaded_parameter_ << "\n";
     ss << "end of parameters" << '\n';
   }
-  return std::move(ss.str());
+  return ss.str();
```
Comment (StrikerRUS) [inline src/boosting/gbdt_model_text.cpp:396]: /gha run-valgrind

Comment (StrikerRUS) [inline src/boosting/gbdt_model_text.cpp:396]: /gha run r-valgrind

Comment (StrikerRUS) [reply src/boosting/gbdt_model_text.cpp:396]: Valgrind is happy: https://github.com/microsoft/LightGBM/actions/runs/323142426.

**Review by guolinke — APPROVED** [review]

File: src/boosting/gbdt_model_text.cpp (hunk 1 above)
Comment (jameslamb) [reply src/boosting/gbdt_model_text.cpp:396]: if valgrind is happy, I am happy 😀

**Pull request merged**